include/seastar/util/later.hh

```diff
@@ -36,6 +36,14 @@ future<> now() {
     return make_ready_future<>();
 }
 
+/// \brief Returns a ready future that is already resolved.
+template<typename T>
+inline
+future<std::remove_cv_t<std::remove_reference_t<T>>> now(T&& v) noexcept {
```

Comment:
If the type isn't movable it could still throw.

Might be worth having an `now<T>(std::in_place, Args&&... args)`?

Reply:
Can you elaborate? I would think you should just use `make_ready_future` if you want in place construction.

Reply:
I don't think there's much value having `noexcept` on a small standalone function with visible body, so yeah it probably makes sense to remove it. The alternative, I guess, is metaprogramming to try to propagate the no-exceptness of the invoked constructor, seems annoying.

Reply:
> If the type isn't movable it could still throw.

Even _if_ is moveable it could throw, right? Also even if a type is "moveable" we will only invoke the move constructor here if we are passed an rvalue, which will probably often not be the case (sometimes in error, i.e., `std::move` was not called where it should have been).

Reply:
Removed the `noexcept` clause.

Reply:
> Can you elaborate? I would think you should just use `make_ready_future` if you want in place construction. Are you suggesting to remove `noexcept` then?

Basically, currently you have to write:
```c++
ss::make_ready_future<T>(T{args...});
```
Honestly, it's a minor point without much of a current demand.

> Removed the `noexcept` clause, thanks all.

I wasn't necessarily arguing against `noexcept` or it's utility, just noting that a copy is possible, and therefore allocs. I don't know if compilers use it as an optimisation, but it might be a `constexpr` limitation.

Reply:
> Basically, currently you have to write:

This seems to work for me:

```c++
struct foo {
    std::string bar;
    size_t qux;
};

int main() {
    auto fut = seastar::make_ready_future<foo>(foo{"nice", 42});
    fut = seastar::now(foo{"nice", 42});
    return 1;
}
```


https://godbolt.org/z/qTsTjfqTj

Reply:
That's what I said? `std::string` is movable.

Reply:
Ah I was misunderstanding. Yeah I think that at least the move constructor is required:

https://github.com/scylladb/seastar/blob/938a4e603ea438d2785eaf6ba1778e6728548e15/include/seastar/core/future.hh#L565-L568

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/performance/noexcept-move-constructor.html

And `ss::make_ready_future` is `noexcept`. So I say leave `noexcewpt` in, maybe at the concept check, but I'm like 2/10 on it.

Reply:
I put `noexcept` back but I not going to mess with the `SEASTAR_CONCEPT` stuff for now.

Reply:
You should ignore my comments above. However I have new better comments. 

I had missed the fact that `ss::make_ready_future` is itself `noexcept` and this is possible because ss::future has a noexecpt constructor (regardless what constructor is called, whether it throws or not), because exceptions during construction are caught and transformed into an exceptional future: see `future_state::future_state(ready_future_marker, A&&... a)` implementation. 

So @BenPope I don't actually see how @rockwotj 's method can throw? In particular, I don't see where an allocation can occur and nothing else that I can see would throw.

Reply:
(so think this should definitely have an unqualified `noexcept` as it does)